### DISCUSSION TITLE
Chicken and the egg scenario with DNS records

### DISCUSSION THREAD

**Discussion by dan-cooke** [pulumi/pulumi#15711] [General]
Hi there,

I'm quite new to configuring DNS. I am using Pulumi to setup my Route53 zones as follows

- example.com -> goes to my production app

Now thats the easy part

But where I am running into trouble is "preview" domains for different stacks for example I want to setup 

- dev.example.com -> NS record in "prod" stack goes to name servers in "dev" stack
- feature-111.example.com -> NS record in "prod" stack goes to name servers in "feature-111" stack

And this seems like a bit of a dependency issue 
```.ts
if (stack === 'prod') {
  const root = new aws.route53.Zone('example.comi', {
    name: 'example.com',
  });
  new aws.route53.Record(`wix-ns`, {
    name: 'example.com',
    zoneId: root.zoneId,
    ttl: 30,
    type: 'NS',
    records: ['ns0.wix.com', 'ns1.wix.com'],
  });

} else {
  // set up subdomain
  const subZone = new aws.route53.Zone(getdomainWithEnvironment(stack), {
    name: getDomainWithEnvironment(stack), // dev.example.com
  });

// now i need to create a NS record in the root zone to point to this sub zone?
// how can i achieve this?
}
```

I can't think of any way of dynamically doing this other than using the management API directly or something

**Comment by Frassle** [comment]
Yeh this is a hard problem. We've got an issue tracking how we could solve this in general: https://github.com/pulumi/pulumi/issues/3021

But for now you're probably best off doing direct calls to the management API to set these up.

**Comment by dan-cooke** [comment]
Oh wow I didn’t think to search for “Chicken & Egg” - that’s interesting
thank you!
